awscli/customizations/cloudformation/deploy.py

```diff
@@ -359,6 +356,68 @@ def merge_parameters(self, template_dict, parameter_overrides):
 
         return parameter_values
 
+    def _validate_cf_params(self, param):
+        if 'ParameterKey' in param and 'ParameterValue' in param:
+            if len(param.keys()) > 2:
+                raise ParamValidationError(
+                    'CloudFormation like parameter JSON should have format '
+                    '[{"ParameterKey": "string", "ParameterValue": "string"}]')
+        return True
+
+    def _cf_param_parser(self, data):
+        # Parse parameter_overrides if they were given in
+        # CloudFormation params file format
+        # [{
+        #   "ParameterKey": "string",
+        #   "ParameterValue": "string",
+        # }]
+        try:
+            return {
+                param['ParameterKey']: param['ParameterValue']
+                for param in data if self._validate_cf_params(param)
+            }
+        except (KeyError, TypeError):
+            return None
+
+    def _codepipeline_param_parser(self, data):
+        # Parse parameter_overrides if they were given in
+        # CodePipeline params file format
+        # {
+        #     "Parameters": {
+        #         "ParameterKey": "ParameterValue"
+        #     }
+        # }
+        if isinstance(data, dict):
+            return data.get('Parameters', None)
+
```

Comment:
Left this off of my last review... Should we validate that it does not have any of the top-level keys like `Tags` and `StackPolicy`? I think it makes sense because the CLI is generally strict on making sure all parameters provided will be used.

Reply:
It's discussible, if we take only Parameters key and just ignore the rest it'll allow customers to reuse the same files without charging